bin/cli.mjs

```diff
@@ -17,145 +18,412 @@ import createMarkdownLoader from '../src/loaders/markdown.mjs';
-  ignore,
-  output,
-  target = [],
-  version,
-  changelog,
-  disableRule,
-  lintDryRun,
-  gitRef,
-  reporter,
-  threads,
-} = program.opts();
-
-const linter = createLinter(lintDryRun, disableRule);
-
-const { loadFiles } = createMarkdownLoader();
-const { parseApiDocs } = createMarkdownParser();
-
-const apiDocFiles = await loadFiles(input, ignore);
-
-const parsedApiDocs = await parseApiDocs(apiDocFiles);
-
-const { runGenerators } = createGenerator(parsedApiDocs);
-
-// Retrieves Node.js release metadata from a given Node.js version and CHANGELOG.md file
-const { getAllMajors } = createNodeReleases(changelog);
-
-// Runs the Linter on the parsed API docs
-linter.lintAll(parsedApiDocs);
-
-if (target) {
-  await runGenerators({
-    // A list of target modes for the API docs parser
-    generators: target,
-    // Resolved `input` to be used
-    input: input,
-    // Resolved `output` path to be used
-    output: output && resolve(output),
-    // Resolved SemVer of current Node.js version
-    version: coerce(version),
-    // A list of all Node.js major versions with LTS status
-    releases: await getAllMajors(),
-    // An URL containing a git ref URL pointing to the commit or ref that was used
-    // to generate the API docs. This is used to link to the source code of the
-    gitRef,
-    // How many threads should be used
-    threads,
-  });
+async function loadAndParse(input, ignore) {
+  const files = await loader.loadFiles(input, ignore);
+  return parser.parseApiDocs(files);
+}
+
+/**
+ * Run the linter on parsed documentation.
+ * @param {ApiDocMetadataEntry[]} docs - Parsed documentation objects.
+ * @param {object} [opts]
+ * @param {string[]} [opts.disableRule] - List of rule names to disable.
+ * @param {boolean} [opts.lintDryRun] - If true, do not throw on errors.
+ * @param {string} [opts.reporter] - Reporter to use for output.
+ * @returns {boolean} - True if no errors, false otherwise.
+ */
+function runLint(
+  docs,
+  { disableRule = [], lintDryRun = false, reporter = 'console' } = {}
+) {
+  const linter = createLinter(lintDryRun, disableRule);
+  linter.lintAll(docs);
+  linter.report(reporter);
+  return !linter.hasError();
+}
+
+/**
+ * Require value to have a length > 0
+ * @param {string} value
+ * @returns {boolean}
+ */
+function requireValue(value) {
+  if (value.length === 0) return 'Value is required!';
+}
+
+/**
+ * Get the message for a prompt
+ * @param {{ message: string, required: boolean }} prompt
+ * @returns {string}
+ */
+function getMessage({ message, required, initialValue }) {
+  return required || initialValue ? message : `${message} (Optional)`;
 }
 
-// Reports Lint Content
-linter.report(reporter);
+/**
+ * Centralized command definitions.
+ * Each command has a description and a set of options with:
+ * - flags: Commander.js flag definitions
+ * - desc: description for help output
+ * - prompt: metadata for interactive mode
+ */
+const commandDefinitions = {
+  generate: {
```

Comment:
Curious if it would be nicer for these to be defined these in separate files (i.e. something like the structure of a [Cobra](https://cobra.dev/#getting-started) app or along the lines of how we define generators)

Reply:
I'm already in the middle of doing that :-). Drafting until I finish